app/assets/javascripts/components/common/feedback.jsx

```diff
@@ -44,6 +45,19 @@ const Feedback = React.createClass({
     this.hide();
   },
 
+  handleFeedbackInputChange(event) {
+    this.setState({ feedbackInput: event.target.value });
+  },
+
+  handleFeedbackSubmit(event) {
+    const feedback = this.state.feedbackInput;
+    this.setState({ feedbackInput: '' });
+    API.createCustomFeedback(this.props.assignment.id, feedback).then(() => {
+      this.props.feedback[this.props.assignment.id].custom.push(<li>{feedback}</li>);
+    });
```

Comment:
This doesn't exactly give the required behavior. If the feedback is stored as a state it will be easier to update but there is no callback to update state after mapStatetoProps (after fetching from API).